internal/rhcs/idps/identity_provider_state.go

```diff
@@ -50,7 +51,8 @@ func IdentityProviderFields() map[string]*schema.Schema {
 			Elem: &schema.Resource{
 				Schema: HtpasswdSchema(),
 			},
-			Optional: true,
+			Optional:     true,
+			ExactlyOneOf: []string{"htpasswd", "gitlab", "github", "google", "ldap", "openid"},
```

Comment:
maybe let's have a constant for ``[]string{"htpasswd", "gitlab", "github", "google", "ldap", "openid"}``?

Reply:
Updated